x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/install_prebuilt_rules_and_timelines/install_prebuilt_rules_package.ts

```diff
@@ -19,24 +22,45 @@ export async function installPrebuiltRulesPackage(
   config: ConfigType,
   context: SecuritySolutionApiRequestHandlerContext
 ) {
-  // Get package version from the config
+  const pkgVersion = await findLatestPackageVersion(config, context, PREBUILT_RULES_PACKAGE_NAME);
+
+  return context
+    .getInternalFleetServices()
+    .packages.ensureInstalledPackage({ pkgName: PREBUILT_RULES_PACKAGE_NAME, pkgVersion });
+}
+
+export async function installEndpointPackage(
+  config: ConfigType,
+  context: SecuritySolutionApiRequestHandlerContext
+) {
+  const pkgVersion = await findLatestPackageVersion(config, context, ENDPOINT_PACKAGE_NAME);
+
+  return context.getInternalFleetServices().packages.ensureInstalledPackage({
+    pkgName: ENDPOINT_PACKAGE_NAME,
+    pkgVersion,
+  });
+}
+
+async function findLatestPackageVersion(
+  config: ConfigType,
+  context: SecuritySolutionApiRequestHandlerContext,
+  packageName: string
+) {
   let pkgVersion = config.prebuiltRulesPackageVersion;
 
   // Find latest package if the version isn't specified in the config
   if (!pkgVersion) {
+    const securityAppClient = context.getAppClient();
     // Use prerelease versions in dev environment
     const isPrerelease =
-      context.getAppClient().getKibanaVersion().includes('-SNAPSHOT') ||
-      context.getAppClient().getKibanaBranch() === 'main';
+      securityAppClient.getBuildFlavor() === 'traditional' &&
+      (securityAppClient.getKibanaVersion().includes('-SNAPSHOT') ||
+        securityAppClient.getKibanaBranch() === 'main');
 
     const result = await context
```

Comment:
Is it not necessary here to check that Fleet is initialized?

Reply:
This is a good question. On the front end, we were waiting for Fleet initialization before installing packages. However, on the backend, similar code simply doesn't exist, and I'm not sure why. I cannot find any initialization attempts or awaiting of initialization completion in the Fleet API. However, they do have a `setup` endpoint that is called from Kibana. I'll try to figure out if we need to call it before installing our packages.